core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -684,6 +684,10 @@ private function addVirtualFilesystemNode(): NodeDefinition
                         ->min(1)
                         ->defaultValue(100)
                     ->end()
+                    ->booleanNode('use_last_modified')
+                        ->info('When enabled, rehashing a file might be skipped if it has not changed since the last time. Note, that the DBAFS service requires a "lastModified" column in the database in this case. All hashes will initially be rewritten to include the file time.')
+                        ->defaultFalse()
```

Comment:
I would argue to switch to `defaultTrue()`. I think we can be fairly sure that the 99% case will be that `filemtime` is actually correct. We're even sort of relying on it for other purposes in Contao e.g. the image thumbnail hashes. So for those who use a filesystem that does not update the last modified date, they'll have a few other places to address and not just this config.

Reply:
There are different places where default values are in play:

* the bundle config (this)
* how it is configured for the core (skeleton config)
* the class itself

My Idea was: with the bundle config defaulting to `false`, you do not have to create a `lastModified` column in your own DBAFS table (minimum is: `id`, `uid`, `pid`, `path`, `hash`, `type` + your own columns). But if we want this to be `true`, I'm fine with it as well. Same reasoning for the class. So: bare minimum + opt-in. These things should probably match for a better DX when creating definitions from scratch.

If we set the core's config (basically `tl_files`) to `true`, we're introducing a change to the database. I.e. the need for recomputing all hashes (migration?) and possibly breaking other peoples code that rely on it. Unlikely but still. :neutral_face:

Reply:
Why not having a general `meta` JSON column for stuff like this? Nobody queries for `last_modified` anyway, do they?

Reply:
> Nobody queries for last_modified anyway, do they?

Hmm, it's then very easy to show the last files that were changed for instance by adding an index + ordering. But it's more that we do not need a JSON field/serialized stuff anymore (I think):

If you want a new storage, say for your company videos with speaker information, you would create a storage definition, enable DBAFS and add a new table `tl_company_videos`, e.g. via a DCA definition. You would probably want to add a column `speaker`,  maybe `language` and so on to the default fields. (If you want/need  last modified, add it as well and enable it.) So every dbafs storage can have its specialized fields without interfering.

Reply:
Then I think it's totally fine to just require this field to be present in the table. The hash update shouldn't be a problem either. No migration needed imho.

Reply:
Discussed this with @ausi. Aaaand the solution is to *not* include the file time in the hash (because it kills the moving detection). I could work around this in another way. And now that there aren't any changes to the hashing needed, I'm absolutely in favor of enabling it by default. :smile:

Reply:
see https://github.com/contao/contao/pull/3774/commits/f8129c5c71399250008a6bf29e8dad8c83b22d01 +  https://github.com/contao/contao/pull/3774/commits/e508cfca00293922378cab8189c12d8e3469e141